add `aptos new` command

mkurnikov opened the pull request:
### Description

Still in WIP state. I want to get early feedback on the subcommand. 

```
aptos new PACKAGE_NAME  
aptos new -t coin PACKAGE_NAME 
aptos new -t dapp PACKAGE_NAME
aptos new -t CUSTOM_URL PACKAGE_NAME 
```

Default case is equivalent-ish to `aptos move init`. 
`-t NAME` fetches templates from https://github.com/mkurnikov/aptos-templates repo. 
`-t CUSTOM_URL` fetches the whole repo. 

All templates allow `{{ package_name }}` template variable (and potentially more in the future). 

TODO:
- [x] Improve `-t TEMPLATE_GIT_URL` work with binary files
- [ ] Better git repository handling
- [ ] Improve aptos-templates

### Test Plan
<!-- Please provide us with clear details for verifying that your changes work. -->

mkurnikov commented:
@movekevin @gregnazario @davidiw

davidiw commented on crates/aptos/src/move_tool/template.rs:
```diff
@@ -0,0 +1,280 @@
+    }
+}
+
+const GIT_APTOS_TEMPLATES_URL: &str = "https://github.com/mkurnikov/aptos-templates.git";
```
this should be in aptos and I think it should be ideally be by branch and maybe commit

mkurnikov replied:
It's already by commit, it works similarly to how dependencies work now.

davidiw reviewed (commented):
thanks for the contribution

let's move this to `aptos move new-package`

should we start with just using a default template that can be inlined into the code that works for everything and then add in other templates? it feels like that adds a bit of complexity and will stall the release of this feature.

mkurnikov commented:
@davidiw 

> should we start with just using a default template that can be inlined into the code that works for everything and then add in other templates

I don't quite understand this comment. 
Currently, if the user wants the default template, like
```
aptos new my_package
```
I am falling back to what `aptos move init` is doing, basically wrapping the call to the command (and creating `tests/` directory later). I never hit the network in the default case. Is this what you suggested?

mkurnikov commented:
I've updated the PR: 
1. Moved command from `aptos new` to `aptos move new-package`.
2. Removed custom template URL functionality. 
3. Moved templates from my custom repo to `aptos-core`. 

Things left:
* do I need to reimplement `aptos move init`? 
* change repo address to aptos-labs instead of my fork

movekevin commented on crates/aptos/src/move_tool/package_templates/coin/sources/user_coin.move:
```diff
@@ -0,0 +1,68 @@
+
+    use aptos_framework::coin::{Self, MintCapability, BurnCapability};
+
+    const ERR_NOT_ADMIN: u64 = 1;
```
Can you add some comments before each error code since these will be returned in the API response when submitting a tx that fails?

movekevin commented on crates/aptos/src/move_tool/template.rs:
```diff
@@ -0,0 +1,226 @@
+use crate::common::types::{CliCommand, CliTypedResult, PromptOptions};
+use crate::move_tool::FrameworkPackageArgs;
+
+const GIT_APTOS_TEMPLATES_URL: &str = "https://github.com/mkurnikov/aptos-core.git";
```
Is this going to be updated to point to the main aptos-core in a separate PR?

mkurnikov replied:
I can do it now, just it won't be possible to test the PR before the merge then.

mkurnikov replied:
UPD. Oh, yes, should be a separate PR.

gregnazario reviewed (commented):
I would rename to `aptos template new`

davidiw commented:
Do we have a design doc or issue we could talk over instead of the PR? It isn't clear to me why there are so many goals in this PR. I personally would have loved to start with a trivial template only and then started working on importing templates. I also think we should agree on the right path. Greg says "aptos template", I think it should be "aptos move new-package", but a PR is an awkward way to have a conversation.
